### PR TITLE
[ci] Match signedness and type in test to avoid warnings in stm32.

### PR DESCRIPTION
**Pull request by strongly-typed**
Never ran `target=stm32` before ...

### CODE REVIEW

**Comment by salkinium** [comment]
Might want to throw in `scons unittest` too ;-P

**Comment by strongly-typed** [comment]
0.

Will you hate me if I accidentally type in `target=atxmega`?

1. 

     xpcc/ui/gui/widgets/widget.cpp:46:1: warning: all paths through this function will call itself [-Winfinite-recursion]

is already mentioned in #177.

2. 

     ./xpcc/communication/xpcc/response_callback.hpp:82:14: warning: 'reinterpret_cast' from class 'TestingComponent2 *' to its base at non-zero offset
      'xpcc::Communicatable *' behaves differently from 'static_cast' [-Wreinterpret-base-class]

Compiler's hint

     note: use 'static_cast' to adjust the pointer correctly while upcasting

is not the solution. It leads to a segfault ..

     ➜  xpcc_feature_release_test git:(fix/release_tests) ✗ lldb ./build/unittest_hosted/executable
     (lldb) target create "./build/unittest_hosted/executable"
     Current executable set to './build/unittest_hosted/executable' (x86_64).
     (lldb) run
     Process 31165 launched: './build/unittest_hosted/executable' (x86_64)
     executable was compiled with optimization - stepping may behave oddly; variables may not be available.
     Process 31165 stopped
     * thread #1: tid = 0x4195d6, 0x000000010001180a executable`xpcc::LinkedList<Timeline::Event,      xpcc::allocator::Dynamic<Timeline::Event> >::append(this=0x0003000000000000, value=<unavailable>) + 48      at linked_list_impl.hpp:104, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (     code=EXC_I386_GPFLT)
         frame #0: 0x000000010001180a executable`xpcc::LinkedList<Timeline::Event, xpcc::allocator::Dynamic<     Timeline::Event> >::append(this=0x0003000000000000, value=<unavailable>) + 48 at      linked_list_impl.hpp:104 [opt]
        101
        102 		// hook the node into the list
        103 		node->next = 0;
     -> 104 		if (this->front == 0)
        105 		{
        106 			// first entry in the list
        107 			this->front = node;

So this might be a tougher one. I do not have enough insight into the xpcc communication judge if the reinterpret cast is a problem.

**Comment by salkinium** [comment]
> Will you hate me if I accidentally type in `target=atxmega`?

I'm spending all my hate on myself for having to forward port all this stuff to modm, nothing left for you. Try again next year.
Keep in mind I'm not reviving Xmega it until after modm.

> is already mentioned in #177.

… and firmly assigned to @daniel-k, who is firmly ignoring me. It's almost like I have no authority over anything anymore. All y'all have no respect! YOU PUNKS GET OFF MY LAWN!

> is not the solution. It leads to a segfault ..

¯\\\_(ツ)\_/¯
cc @dergraaf

File: src/xpcc/architecture/platform/board/nucleo_l476rg/nucleo_l476rg.hpp [hunk 1]
```diff
@@ -87,6 +87,9 @@ using Rx = GpioInputA3;
 using Uart = Usart2;
 }
 
+// Create an IODeviceWrapper around the Uart Peripheral we want to use
+using LoggerDevice = xpcc::IODeviceWrapper< Board::stlink::Uart, xpcc::IOBuffer::BlockIfFull >;
+
```
Comment (salkinium) [inline src/xpcc/architecture/platform/board/nucleo_l476rg/nucleo_l476rg.hpp]: Should we add this to all boards then?

**Review by salkinium — APPROVED** [review]

**Comment by salkinium** [comment]
I may squash the commits down to maybe 2-3 commits, as this is a bit verbose.

**Comment by salkinium** [comment]
Thanks for the good work!

**Pull request merged**